common/djangoapps/student/models.py

```diff
@@ -2363,6 +2363,15 @@ class Meta(object):
     )
 
 
+class SocialLink(models.Model):  # pylint: disable=model-missing-unicode
+    """
+    Represents a URL connecting a particular social platform to a user's social profile.
+    """
+    user_profile = models.ForeignKey(UserProfile, db_index=True, related_name='social_links')
+    platform = models.CharField(max_length=30)
+    social_link = models.CharField(max_length=100, blank=True)
```

Comment:
Good point, I'll add a comment to this model explaining the necessary parts. It's tricky to add the choices fields to the model because they can always change if the settings file is altered in any way.